If the system environment variables include `kafka` or `KAFKA`, the `KafkaRuntimeConfigProducer` throws an exception.

### Describe the bug

If the system environment variables include `kafka` or `KAFKA`, the `KafkaRuntimeConfigProducer` throws an error.
```
Caused by: java.lang.StringIndexOutOfBoundsException: Range [6, 5) out of bounds for length 5
	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:55)
	at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:52)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:213)
	at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:210)
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckFromToIndex(Preconditions.java:112)
	at java.base/jdk.internal.util.Preconditions.checkFromToIndex(Preconditions.java:349)
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4861)
	at java.base/java.lang.String.substring(String.java:2830)
	at java.base/java.lang.String.substring(String.java:2803)
	at io.quarkus.kafka.client.runtime.KafkaRuntimeConfigProducer.createKafkaRuntimeConfig(KafkaRuntimeConfigProducer.java:40)
	at io.quarkus.kafka.client.runtime.KafkaRuntimeConfigProducer_ProducerMethod_createKafkaRuntimeConfig_VzcC8CdA4ZIN2S227pAbEVHVF_c_Bean.doCreate(Unknown Source)
	at io.quarkus.kafka.client.runtime.KafkaRuntimeConfigProducer_ProducerMethod_createKafkaRuntimeConfig_VzcC8CdA4ZIN2S227pAbEVHVF_c_Bean.create(Unknown Source)
	at io.quarkus.kafka.client.runtime.KafkaRuntimeConfigProducer_ProducerMethod_createKafkaRuntimeConfig_VzcC8CdA4ZIN2S227pAbEVHVF_c_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:119)
```

The `substring` here throws an exception:
https://github.com/quarkusio/quarkus/blob/670b1ad3c991bcbd644cb04290eb1d3b3688bf3c/extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/KafkaRuntimeConfigProducer.java#L40-L41

### Expected behavior

Work without throwing any error

### Actual behavior

throw exception: `Caused by: java.lang.StringIndexOutOfBoundsException: Range [6, 5) out of bounds for length 5`

### How to Reproduce?

1. Configuring the system environment variable KAFKA=xxx.xxx.xxx.xxx:9092.
2. Start the kafka application

### Output of `uname -a` or `ver`

Windows_NT minij 6.2 9200 x86_64 MS/Windows

### Output of `java -version`

openjdk version "21" 2023-09-19 OpenJDK Runtime Environment (build 21+35-2513) OpenJDK 64-Bit Server VM (build 21+35-2513, mixed mode, sharing)

### Quarkus version or git rev

3.10.1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)

### Additional information

_No response_

--- a/extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/KafkaRuntimeConfigProducer.java
+++ b/extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/KafkaRuntimeConfigProducer.java
@@ -35,6 +35,9 @@ public class KafkaRuntimeConfigProducer {
             if (!propertyNameLowerCase.startsWith(CONFIG_PREFIX) || propertyNameLowerCase.startsWith(UI_CONFIG_PREFIX)) {
                 continue;
             }
+            if (propertyNameLowerCase.length() <= CONFIG_PREFIX.length()) {
+                continue;
+            }
             // Replace _ by . - This is because Kafka properties tend to use . and env variables use _ for every special
             // character. So, replace _ with .
             String effectivePropertyName = propertyNameLowerCase.substring(CONFIG_PREFIX.length() + 1).toLowerCase()

